calicoctl/commands/edit.go

```diff
@@ -174,9 +169,20 @@ Description:
 
 		// Convert the file into a set of resources.  If this succeeds
 		// then it passes validation, so exit this loop.
+		var editedObjs []unversioned.ResourceObject
+		var prevObjs []unversioned.ResourceObject
 		resources, err = resourcemgr.CreateResourcesFromBytes(b)
 		if err == nil {
```

Comment:
I think it would be easier to invert this logic to reduce the nesting.

Better I think now to check `if err != nil` to log a specific log and then `continue` the loop, eg.

```
if resources, err = resourcemgr.CreateResourcesFromBytes(b); err != nil {
    log
    continue
} else if editedObjs, err = convertToSliceOfResourceObjects(resources); err != nil {
    log
    continue
} else ...
```